Makefile

```diff
@@ -12,6 +12,22 @@ CONTAINER_IMAGE   = $(IMAGE_REGISTRY)/k8s-website-hugo:v$(HUGO_VERSION)-$(IMAGE_
 # Mount read-only to allow use with tools like Podman in SELinux mode
 # Container targets don't need to write into /src
 CONTAINER_RUN     = "$(CONTAINER_ENGINE)" run --rm --interactive --tty --volume "$(CURDIR):/src:ro,Z"
+CONTAINER_RUN_TTY = "$(CONTAINER_ENGINE)" run --rm --interactive --tty
+CONTAINER_HUGO_MOUNTS = \
+	--read-only \
+	--mount type=bind,source=$(CURDIR)/.git,target=/src/.git,readonly \
+	--mount type=bind,source=$(CURDIR)/archetypes,target=/src/archetypes,readonly \
+	--mount type=bind,source=$(CURDIR)/assets,target=/src/assets,readonly \
+	--mount type=bind,source=$(CURDIR)/content,target=/src/content,readonly \
+	--mount type=bind,source=$(CURDIR)/data,target=/src/data,readonly \
+	--mount type=bind,source=$(CURDIR)/i18n,target=/src/i18n,readonly \
+	--mount type=bind,source=$(CURDIR)/layouts,target=/src/layouts,readonly \
+	--mount type=bind,source=$(CURDIR)/public,target=/src/public \
+	--mount type=bind,source=$(CURDIR)/resources,target=/src/resources \
+	--mount type=bind,source=$(CURDIR)/static,target=/src/static,readonly \
+	--mount type=bind,source=$(CURDIR)/themes,target=/src/themes,readonly \
+	--mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 \
+	--mount type=bind,source=$(CURDIR)/hugo.toml,target=/src/hugo.toml,readonly
```

Comment:
The switch from "the build happens in a container" to "the app can write to the repo" is possibly going to break things; it does for me.

I'm going to see if any of these can be tmpfs (eg `/src/public`) rather than a mount in.

Reply:
Are you referringt to writing to `public`? That is how it is already handled in `main`: https://github.com/kubernetes/website/blob/a371603439fde7d34e329e3f88e12e7c5fba5645/Makefile#L99-L104

Can you not build `main`? If you can't run a containerized build from main, then we should create a separate issue to track and fix that. Specifically what is failing?

Reply:
It's too bad that writing to `public` doesn't work for you. It would give folks doing containerized builds the ability to see the generated site files.